src/interface.jl

```diff
@@ -685,15 +685,15 @@ function promote_operation_fallback(
 end
 
 function buffer_for(::typeof(add_dot), a::Type, b::Type, c::Type)
-    return buffer_for(add_mul, a, promote_operation(adjoint, b), c)
+    return buffer_for(add_mul, a, promote_operation(transpose, b), c)
 end
 
 function operate_to_fallback!(::IsMutable, output, ::typeof(add_dot), a, b, c)
-    return operate_to!(output, add_mul, a, adjoint(b), c)
+    return operate_to!(output, add_mul, a, transpose(b), c)
```

Comment:
Ah. The kicker is here. These methods actually fallback to `add_mul`, which uses `*` instead of `dot` for the element-wise operations.